src/Data/Fin/Properties.agda

```diff
@@ -1008,8 +1008,25 @@ module _ {f} {F : Set f → Set f} (RF : RawFunctor F) where
 -- If there is an injection from a type A to a finite set, then the type
 -- has decidable equality.
 
+module FromInjection {n a e} {A : Setoid a e}
+                     (inj : Injection A (≡-setoid n)) where
```

Comment:
Hmm so we don't usually use named inner modules, and if we do we always open them publicly afterwards. The definition of `FromInjection.eq?` is compatible with `eq` right?  `eq?` is also a terrible name for the existing property. So how about we deprecate `eq?` and then add the following:

```agda
module _ {ℓ} {S : Setoid a ℓ} (inj : Injection S (≡-setoid n)) where
  
  inj⇒≟ : -> B.Decidable _≈_
  inj⇒≟ inj = Dec.via-injection inj _≟_

  inj⇒decSetoid : DecSetoid a e
  inj⇒decSetoid = ...
``

Reply:
Done.